gammapy/cube/make.py

```diff
@@ -303,10 +332,39 @@ def _make_background(self):
 
         self.maps["background"] = background
 
+    def _make_edisp(self):
+
+        energy_axis = self.geom_true.get_axis_by_name("ENERGY")
+        geom_migra = self.geom_true.to_image().to_cube([self.migra_axis, energy_axis])
+        edisp_map = make_edisp_map(
+            edisp=self.observation.edisp,
+            pointing=self.observation.pointing_radec,
+            geom=geom_migra,
+            max_offset=self.offset_max,
+            exposure_map=self.maps["exposure_irf"],
+        )
+        self.maps["edisp"] = edisp_map
+
+    def _make_psf(self):
+        # CTA 1DC psf is gammapy.irf.psf_gauss.EnergyDependentMultiGaussPSF
+        # TODO: correct handling for 1DC data?
+        if type(self.observation.psf) is not PSF3D:
```

Comment:
For this kind of checks one typically uses `if isinstance(..., PSF3D)`. Note that there is https://github.com/gammapy/gammapy/blob/master/gammapy/irf/psf_gauss.py#L451, which allows you to transform to a `PSF3D`.

Reply:
Thanks! I add a conversion now.